lib/Parse/ParseStmt.cpp

```diff
@@ -368,6 +368,25 @@ ParserStatus Parser::parseBraceItems(SmallVectorImpl<ASTNode> &Entries,
       diagnose(EndOfPreviousLoc, diag::statement_same_line_without_semi)
         .fixItInsert(EndOfPreviousLoc, ";");
       // FIXME: Add semicolon to the AST?
+      Token previousToken = Lexer::getTokenAtLocation(SourceMgr, PreviousLoc);
+      Optional<tok> stmtTokMatch = None;
+      if (previousToken.getKind() == tok::identifier) {
+        SourceLoc startOfLineLocation = L->getLocForStartOfLine(SourceMgr, previousToken.getLoc());
```

Comment:
Also @rintaro if we are trying to use simple skipping solution, should we remove this check for start of the line. Example:

func foo() -> Int? { return 0 }
func bar() {
  let i = 5; gaurd let x = foo() else { return }  // expected-error {{consecutive statements}} // expected-note {{did you misspell 'guard'}} {{3-8=guard}}
  print(x)
}
Should we consider it ?

Reply:
For now, how about just skipping the first `Tok.isAny(tok::kw_let, tok::kw_var)` in this recovery logic? like:
```
if (Tok.isAny(tok::kw_let, tok::kw_var))
  consumeToken();
skipUntilDeclStmtRBrace()
```
It's not great for something like:
```swift
gurad
  let a = someOpt1,
  let b = someOpt2
else {
}
```
But I think supporting single `let`/`var` is good enough because this is just a recovery.

Reply:
@rintaro Sure. Updated.